plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchMetadata.java

```diff
@@ -365,10 +365,12 @@ else if (type instanceof ScaledFloatType) {
             return new TypeAndDecoder(DOUBLE, new DoubleDecoder.Descriptor(path));
         }
         else if (type instanceof DateTimeType) {
-            if (((DateTimeType) type).getFormats().isEmpty()) {
-                return new TypeAndDecoder(TIMESTAMP_MILLIS, new TimestampDecoder.Descriptor(path));
+            List<String> formats = ((DateTimeType) type).getFormats();
+            String pattern = null;
+            if (!formats.isEmpty()) {
+                pattern = String.join("||", formats);
```

Comment:
There's no test coverage taking several formats and joining them.

Reply:
We need to index more than 2 documents to test the case of multiple formats. Current `index()` function for testing does not support this. I need to update this function, should I include the feature in this PR?